code/__HELPERS/mob_helpers.dm

```diff
@@ -455,6 +455,59 @@
 	if(progress)
 		qdel(progbar)
 
+/// A version of do_after that will not get cancelled by a mob moving, for situations when signals are not viable
+/proc/moving_do_after(mob/user, delay, needhand = TRUE, atom/target = null, progress = TRUE, list/extra_checks = list(), use_default_checks = TRUE)
```

Comment:
I'd rather have an extra param on `do_after`/`do_after_once` instead which does this. Else we'll have even more `do_after` versions (looking at you `do_mob`).
`do_mob` already allows the stuff you want btw. In an odd way though, take a look at how the `/obj/effect/proc_holder/spell/charge_up/proc/StartChargeup` proc handles it.